iacrcc/iacrcc.cls

```diff
@@ -144,72 +144,112 @@
   \IfFileExists{luatex85.sty}{\RequirePackage{luatex85}}{}
 \fi
 
-% The hyperref package is used to handle cross-referencing commands in 
-% LaTeX to produce hypertext links in the document. hyperref is fussy about
-% the order in which it is loaded, and the official documentation recommends
-% loading last. In spite of this, there are some packages that must be
-% loaded before hyperref. (e.g., cleveref). In order to support this, the
-% author should create a file called before-hyperref.sty with
-% \RequirePackage{longtable} in it.
-\IfFileExists{before-hyperref.sty}{%
-  \RequirePackage{before-hyperref}
-}{}
-
-% XMP (eXtensible Metadata Platform) is a mechanism proposed by Adobe
-% for embedding document metadata within the document itself. It should
-% be loaded before hyperref. We require a version later than 2019/03/14,
-% and this will issue a warning if that is not satisfied.
-\RequirePackage{hyperxmp}[2019/03/14]
 \setcounter{tocdepth}{2}
 
-\RequirePackage{hyperref}
-\hypersetup{pdflang=en,
-            urlcolor=blue,
-            colorlinks=true,
-            citecolor=black!70!green,
-            linkcolor=black!70!red}
-% Disable latexdiff commands in PDF links
-\pdfstringdefDisableCommands{%
-  \def\DIFadd#1{#1}%
-  \def\DIFdel#1{}%
-}
 % doclicense added support for the macro \doclicenseLongTextForHyperref in version 3.1.0 (2022/04/27)
 \@ifpackagelater{doclicense}{2022/04/25}{\@doclicense@LongTextForHyperreftrue}{}
 \@ifpackageloaded{biblatex}{}{\bibliographystyle{alphaurl}}
+
 \AtEndPreamble{%
   \ifpdftex
     % TU encoding is only available with XeTeX and LuaTeX.
     % Defaulting to T1 encoding.
+    % fontenc: Standard package for selecting font encodings
     \RequirePackage[T1]{fontenc} 
   \fi
-  %% PDF title and author
+  \ifpdftex
```

Comment:
I'm not sure why the load order depends on whether we are using pdflatex or not.

Reply:
There is another potential problem. Whenever the author calls `\license`, this will cause `doclicense` to be loaded, which in turn will load `hyperxmp`. If they call it in the preamble, then it's before the end of the preamble. For this reason I think we should remove the loading of `doclicense` in the `\license` macro. We would instead just set the value of the license there, and in the `\AtEndPreamble` we load doclicense with the correct license. I think you can either check that the license is valid in `\license` or in `\AtEndPreamble`.